### PR TITLE
Improve Testing

### PR DESCRIPTION
**Pull request by slider1337**
This will improve the test coverage and also fit the header, put, and patch methods.
There is some BC-Breaks:
-  GetRequestRouter: the route Method not gets a HttpRequestInterface passed, it is also used for header request
- Header Request were CommandRequests, but are now handled like GET request, because the nature of get request is more fitting

Futher changes:
- update phpunit and composer
- add docker image for running tests
- add unit tests for form population, Url and a example kernel test empty template
- add missing NotFoundPage content
- fix workflow for put, patch, header
- add example kernel to test the application workflow

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -73,10 +72,10 @@ vendor/bin/zeus c:a Simovative\\Demo Demo
 This will create a folder public with an index.php file and a folder "bundles" 
 containing the ApplicationBundle and ApplicationKernel.
 
-You might want to configure your webserver right away to test if this
+You might want to configure your web-server right away to test if this
```
Comment (diecktator) [inline Readme.md]: I think you would just write web server, without a dash

File: Readme.md [hunk 2]
```diff
@@ -73,10 +72,10 @@ vendor/bin/zeus c:a Simovative\\Demo Demo
...
 worked. 
 
-### Webserver config
+### Web-Server config
```
Comment (diecktator) [inline Readme.md]: see above

File: Readme.md [hunk 3]
```diff
@@ -96,7 +95,7 @@ location / {
 }
 ```
 #### PHP internal webserver
-Just run the webserver and use the public/index.php as router script:
+Just run the web-server and use the public/index.php as router script:
```
Comment (diecktator) [inline Readme.md]: see above

File: Readme.md [hunk 4]
```diff
@@ -165,10 +164,13 @@ vendor/bin/zeus c:c Login Test
...
 * LoginCommand.php: The command object that holds all the data needed for execution.
 * LoginCommandBuilder.php The command builder connects all the command classes 
-together and will be used by the command dispatcher to get the single components.
+and will be used by the command dispatcher to get the single components.
 * LoginCommandHandler.php: The handler does the command execution.
-* LoginCommandValidator.php: The validator checks the command request if all the data
-is there and valid that is needed for the command execution.
+* LoginCommandValidator.php: The validator checks the command request if the provided data is valid to create
+a command and execute the handler. We recommend basic checks for required data and if the data types are correct and
+data is in the correct format (e. g. like e-mail addresses). If the command can not be successfully executed because of
```
Comment (diecktator) [inline Readme.md]: e. g. => e.g.

File: tests/Unit/Http/Post/UploadedFileTest.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
+<?php
```
Comment (diecktator) [inline tests/Unit/Http/Post/UploadedFileTest.php]: formatting is very strange in this file
